### ISSUE TITLE
How to get the list of broken links and unused files?

### ISSUE THREAD

**Issue by rei-vilo** [mkdocs/mkdocs#1765]
When launched, `mkdocs build` reports unused pages and missing files.

```
INFO    -  Cleaning site directory 
INFO    -  Building documentation to directory: ~/MkDocs/site 
INFO    -  The following pages exist in the docs directory, 
           but are not included in the "nav" configuration:
  - Hidden Page.md 
WARNING -  Documentation file 'Chapter1/Section2.md' contains a link to 
           'Chapter1/img/MissingFile.png' which is not found in the documentation files. 
```

Is there a way to get the list of broken links and unused files?

I tried `mkdocs -v build` to no avail.

Thank you!

**Comment by waylan** [comment]
I'm confused. The list of pages are included in the message you posted above. Is there some other list you are looking for?

Specifically, the "unused pages":

```
INFO    -  The following pages exist in the docs directory, 
           but are not included in the "nav" configuration:
  - Hidden Page.md 
```

The list is at the end of that message and contains one item: `Hidden Page.md `.

And for pages with broken links:

```
WARNING -  Documentation file 'Chapter1/Section2.md' contains a link to 
           'Chapter1/img/MissingFile.png' which is not found in the documentation files.
```

Again, the information is right in the warning message, both the file with the bad link, and the bad link itself. Not sure what else you need.

**Comment by rei-vilo** [comment]
Thank you for your answer. Yes, it isn't easy to explain. Let me rephrase my question:

Today, `mkdocs` mentions 

- the pages that exist but are not listed in the `mkdocs.yml`, 

```
INFO    -  The following pages exist in the docs directory, but are not included in the "nav" configuration:
  - Hidden Page.md 
```

- the pages listed on `mkdocs.yml` the but do not exist, and

```
WARNING -  A relative path to 'Chapter1/Missing.md' is included in the 'nav' configuration, which is not found in the documentation files 
```

- the links that point to a missing page.

```
WARNING -  Documentation file 'Chapter1/Section1.md' contains a link to 'Chapter1/Missing.md' which is not found in the documentation files. 
```

So all work fine for the documents `.md`.

However, all the images, used or not, are copied into the `site` folder. 

I would like to have the list of the files, _e.g._ images, that are not used by the documents `.md`.

As an alternative, I would like to have on the `site` folder the images that are actually used by the documents `.md`. 

The idea is to save space and bandwidth!

**Comment by waylan** [comment]
> I would like to have the list of the files, e.g. images, that are not used by the documents `.md`.

This is not tracked in any way.

**Comment by wilhelmer** [comment]
**Tip:** If you are running Windows and just need a list of unused files in your project (no automation), [this](https://www.codeproject.com/Articles/555489/Find-Unused-Files-v) is an excellent tool. Simply point it to the MkDocs output dir, e.g., ```/site```.

**Comment by wilhelmer** [comment]
@rei-vilo Three years later, I've developed a plugin that does this: [mkdocs-unused-files](https://github.com/wilhelmer/mkdocs-unused-files). Better late than never 😄

**Comment by rei-vilo** [comment]
Thank you! I am going to try it and let you know.

**Comment by ssbarnea** [comment]
@wilhelmer I would quite cool if your plugin could detect broken links.

**Comment by wilhelmer** [comment]
@ssbarnea That's what [htmlproofer](https://github.com/manuzhang/mkdocs-htmlproofer-plugin) is for. 

Use both unused-files and htmlproofer for maintenance runs 👍
